shotover-proxy/src/message/mod.rs

```diff
@@ -104,6 +104,16 @@ impl Message {
             },
         }
     }
+
+    /// Gets the `QueryType` for a message.
+    /// First checks the `details` of the message and then falls back to the `original` message.
+    pub fn get_query_type(&self) -> QueryType {
+        if let MessageDetails::Query(query) = &self.details {
+            query.query_type.clone()
```

Comment:
We can avoid this clone by borrowing since we are only using the result for equality checks. It's only an enum though so not going to block on this.

Reply:
This is a very simple enum without any fields so cloning here should be as cheap as returning an i32.
I should probably implement Copy for QueryType to remove the `clone()` to make it clear that copying it is a cheap operation. AFAIK that would compile down to the same thing but it would just make it look less performance-scary.

Reply:
I'll resolve this and merge, we can do such a change in a separate PR later.